### PR TITLE
Added basic implementation

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR add a basic implementation for the activity-log.

### CODE REVIEW

File: src/Activity/Activity.php [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\ActivityLog\Activity;
```
Comment (chirimoya) [inline src/Activity/Activity.php]: `namespace Sulu\Component\ActivityLog\Model;`

File: src/Activity/Activity.php [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Container for activity data.
+ */
+class Activity implements ActivityInterface
```
Comment (chirimoya) [inline src/Activity/Activity.php]: `ActivityLog`

File: src/Activity/ActivityInterface.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+namespace Sulu\Component\ActivityLog\Activity;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Interface for activities.
+ */
+interface ActivityInterface
```
Comment (chirimoya) [inline src/Activity/ActivityInterface.php]: `ActivityLogInterface `

File: src/ActivityLogger.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log(ActivityInterface $activity)
+    {
+        $this->newActivities[] = $activity;
+        $this->storage->persist($activity);
+
+        $this->eventDispatcher->dispatch(Events::PRE_LOG_ACTIVITY_EVENT, new ActivityEvent($activity));
```
Comment (chirimoya) [inline src/ActivityLogger.php]: `Events::PERSIST_ACTIVITY_LOG_EVENT`

File: src/ActivityLogger.php [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function flush()
+    {
+        $this->storage->flush();
+
+        foreach ($this->newActivities as $activity) {
+            $this->eventDispatcher->dispatch(Events::POST_LOG_ACTIVITY_EVENT, new ActivityEvent($activity));
```
Comment (chirimoya) [inline src/ActivityLogger.php]: `Events::FLUSH_ACTIVITY_LOG_EVENT` with all new activities

File: src/Events/ActivityEvent.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Sulu\Component\ActivityLog\Events;
+
+use Sulu\Component\ActivityLog\Activity\ActivityInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event-Args for activity events.
+ */
+class ActivityEvent extends Event
```
Comment (chirimoya) [inline src/Events/ActivityEvent.php]: `ActivityLogEvent `

File: src/Events/ActivityEvent.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+{
+    /**
+     * @var ActivityInterface
+     */
+    protected $activity;
```
Comment (chirimoya) [inline src/Events/ActivityEvent.php]: `$activityLog`

File: src/ActivityLogger.php [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+     */
+    public function findByParent(ActivityInterface $activity, $page = 1, $pageSize = null)
+    {
+        return $this->storage->findByParent($activity, $page, $pageSize);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log(ActivityInterface $activity)
```
Comment (chirimoya) [inline src/ActivityLogger.php]: `persist`

File: src/Events/Events.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+namespace Sulu\Component\ActivityLog\Events;
+
+/**
+ * Container for activity event-names.
+ */
+final class Events
+{
+    const PRE_LOG_ACTIVITY_EVENT = 'sulu_activity.pre_log';
```
Comment (chirimoya) [inline src/Events/Events.php]: const PERSIST_ACTIVITY_LOG_EVENT = 'sulu_activity_log.persist';

File: src/Storage/ActivityStorageInterface.php [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+namespace Sulu\Component\ActivityLog\Storage;
+
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+
+/**
+ * Interface for activity-storage.
+ */
+interface ActivityStorageInterface
```
Comment (chirimoya) [inline src/Storage/ActivityStorageInterface.php]: `ActivityLogStorageInterface `

File: src/ActivityLoggerInterface.php [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+
+/**
+ * Interface for activity-logger.
+ */
+interface ActivityLoggerInterface
+{
+    /**
+     * Returns activity by given uuid.
```
Comment (chirimoya) [inline src/ActivityLoggerInterface.php]: Returns activity log by given uuid.

File: src/ActivityLoggerInterface.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+     * @param ActivityLogInterface $activityLog
+     * @param int $page
+     * @param int $pageSize
+     *
+     * @return ActivityLogInterface[]
+     */
+    public function findByParent(ActivityLogInterface $activityLog, $page = 1, $pageSize = null);
+
+    /**
+     * Stores activity.
```
Comment (chirimoya) [inline src/ActivityLoggerInterface.php]: Stores activity log.

File: src/Events/Events.php [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\ActivityLog\Events;
+
+/**
+ * Container for activity event-names.
```
Comment (chirimoya) [inline src/Events/Events.php]: Container for activity log event-names.

File: src/Events/FlushActivityLogEvent.php [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\ActivityLog\Events;
+
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event-Args for activity persist-event.
```
Comment (chirimoya) [inline src/Events/FlushActivityLogEvent.php]: Event-Args for activity log flush-event.

File: src/Events/PersistActivityLogEvent.php [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\ActivityLog\Events;
+
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event-Args for activity persist-event.
```
Comment (chirimoya) [inline src/Events/PersistActivityLogEvent.php]: Event-Args for activity log persist-event.

File: src/Events/PersistActivityLogEvent.php [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * @param ActivityLogInterface $activityLog
+     */
+    public function __construct(ActivityLogInterface $activityLog)
+    {
+        $this->activityLog = $activityLog;
+    }
+
+    /**
+     * Returns activity.
```
Comment (chirimoya) [inline src/Events/PersistActivityLogEvent.php]: Returns activity log.

File: src/Storage/ActivityStorageInterface.php [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @param int $page
+     * @param int $pageSize
+     *
+     * @return ActivityLogInterface[]
+     */
+    public function findAll($page = 1, $pageSize = null);
+
+    /**
+     * Find activities by given parent activity.
```
Comment (chirimoya) [inline src/Storage/ActivityStorageInterface.php]: Find activity logs by given parent activity log.

File: src/Storage/ActivityStorageInterface.php [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+     * @param ActivityLogInterface $activityLog
+     * @param int $page
+     * @param int $pageSize
+     *
+     * @return ActivityLogInterface
+     */
+    public function findByParent(ActivityLogInterface $activityLog, $page = 1, $pageSize = null);
+
+    /**
+     * Persists activity.
```
Comment (chirimoya) [inline src/Storage/ActivityStorageInterface.php]: Persists activity log.

File: src/Model/ActivityLog.php [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\ActivityLog\Model;
+
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Container for activity data.
```
Comment (chirimoya) [inline src/Model/ActivityLog.php]: Container for activity log data.

File: src/Storage/ActivityStorageInterface.php [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+
+/**
+ * Interface for activity-storage.
+ */
+interface ActivityStorageInterface
+{
+    /**
+     * Find activity by uuid.
```
Comment (chirimoya) [inline src/Storage/ActivityStorageInterface.php]: Find activity log by uuid.

File: src/Storage/ArrayStorage/ArrayActivityStorage.php [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+
+namespace Sulu\Component\ActivityLog\Storage\ArrayStorage;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sulu\Component\ActivityLog\Model\ActivityLogInterface;
+use Sulu\Component\ActivityLog\Storage\ActivityStorageInterface;
+
+/**
+ * Implementation of activity-storage using doctrine-collection.
```
Comment (chirimoya) [inline src/Storage/ArrayStorage/ArrayActivityStorage.php]: Implementation of activity-log storage using doctrine-collection.

File: tests/Functional/ActivityLoggerTest.php [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+        $this->collection->add(ActivityLog::create('default'));
+
+        $this->assertCount(3, $this->logger->findAll());
+        $this->assertCount(2, $this->logger->findAll(1, 2));
+        $this->assertCount(1, $this->logger->findAll(2, 2));
+    }
+
+    public function testFindByParent()
+    {
+        $parentActivity = ActivityLog::create('default');
```
Comment (chirimoya) [inline tests/Functional/ActivityLoggerTest.php]: `$parentActivityLog`

File: tests/Unit/Storage/ArrayStorage/ArrayActivityStorageTest.php [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+        $activityLog = ActivityLog::create('default');
+        $storage = new ArrayActivityStorage();
+
+        $this->assertEquals($activityLog, $storage->persist($activityLog));
+        $this->assertEquals([$activityLog], $storage->findAll());
+    }
+
+    public function testPersistWithParent()
+    {
+        $parentActivity = ActivityLog::create('default');
```
Comment (chirimoya) [inline tests/Unit/Storage/ArrayStorage/ArrayActivityStorageTest.php]: `$parentActivityLog`

**Comment by wachterjohannes** [comment]
@chirimoya i hope that now all the renamings are done (:

**Pull request merged**